cognite_toolkit/_cdf_tk/warnings.py

```diff
@@ -52,3 +64,40 @@ def __str__(self) -> str:
             for warning in group_list:
                 output.append(f"{'    ' * 2} * {warning!s}")
         return "\n".join(output)
+
+
+@dataclass(frozen=True)
+class GeneralWarning(ToolkitWarning):
```

Comment:
Maybe making this an abstract class to signal it is not intended to be used, but instead derived classes.
```suggestion
class GeneralWarning(ToolkitWarning, ABC):
```